test/ArchToken.js

```diff
@@ -0,0 +1,67 @@
+// We import Chai to use its asserting functions here.
+const { expect } = require("chai");
+const exp = require("constants");
+
+describe("Arch Token test suit", function () {
+    // parseUnits uses 18 decimal by default
+    const tokenSupply = ethers.utils.parseUnits("100");
+    console.log(tokenSupply);
+
+    let contract;
+    let token;
+    let owner;
+    let user1;
+    let user2;
+    let users;
+
+    beforeEach(async function () {
+        contract = await ethers.getContractFactory("ArchToken");
+        [owner, user1, user2, ...users] = await ethers.getSigners();
+        token = await contract.deploy();
```

Comment:
Worth checking if pre-mint worked and minted the correct amount in the `constructor`

Reply:
Added